plugins/flytekit-mlflow/flytekitplugins/mlflow/tracking.py

```diff
@@ -0,0 +1,127 @@
+    """
+    Extracts all metrics and returns a dictionary of metric name to the list of metric for the given run_id
+    """
+    r = c.get_run(run_id)
+    metrics = {}
+    for k in r.data.metrics.keys():
+        metrics[k] = metric_to_df(metrics=c.get_metric_history(run_id=run_id, key=k))
+    return metrics
+
+
+def get_run_params(c: MlflowClient, run_id: str) -> typing.Optional[pd.DataFrame]:
+    """
+    Extracts all parameters and returns a dictionary of metric name to the list of metric for the given run_id
+    """
+    r = c.get_run(run_id)
+    name = []
+    value = []
+    if r.data.params == {}:
+        return None
+    for k, v in r.data.params.items():
+        name.append(k)
+        value.append(v)
+    return pd.DataFrame(list(zip(name, value)), columns=["name", "value"])
+
+
+def plot_metrics(metrics: typing.Dict[str, pandas.DataFrame]) -> typing.Optional[go.Figure]:
+    v = len(metrics)
+    if v == 0:
+        return None
+
+    # Initialize figure with subplots
+    fig = make_subplots(rows=v, cols=1, subplot_titles=list(metrics.keys()))
+
+    # Add traces
+    row = 1
+    for k, v in metrics.items():
+        v["timestamp"] = (v["timestamp"] - v["timestamp"][0]) / 1000
+        fig.add_trace(go.Scatter(x=v["timestamp"], y=v["value"], name=k), row=row, col=1)
+        row = row + 1
+
+    fig.update_xaxes(title_text="Time (s)")
+    fig.update_layout(height=700, width=900)
+    return fig
+
+
+def mlflow_autolog(fn=None, *, framework=mlflow.sklearn, experiment_name: typing.Optional[str] = None):
+    """
+    This decorator can be used as a nested decorator for a ``@task`` and it will automatically enable mlflow autologging,
+    for the given ``framework``. If framework is not provided then the autologging is enabled for ``sklearn``
+    .. code-block::python
+        @task
+        @mlflow_autolog(framework=mlflow.tensorflow)
+        def my_tensorflow_trainer():
+            ...
+    One benefit of doing so is that the mlflow metrics are then rendered inline using FlyteDecks and can be viewed
+    in jupyter notebook, as well as in hosted Flyte environment
+    .. code-block:: python
+        with flytekit.new_context() as ctx:
+            my_tensorflow_trainer()
+            ctx.get_deck()  # IPython.display
+    The decorator starts a new run, with mlflow for the task
+    """
+
+    @wraps(fn)
+    def wrapper(*args, **kwargs):
+        framework.autolog()
+        params = FlyteContextManager.current_context().user_space_params
+        ctx = FlyteContextManager.current_context()
+
+        experiment = experiment_name or "local workflow"
+        run_name = None  # MLflow will generate random name if value is None
+
+        if ctx.execution_state.mode != ExecutionState.Mode.LOCAL_WORKFLOW_EXECUTION:
+            experiment = f"{get_one_of('FLYTE_INTERNAL_EXECUTION_WORKFLOW', '_F_WF')}"
+            run_name = f"{params.execution_id.name}.{params.task_id.name.split('.')[-1]}"
```

Comment:
should this use `experiment_name` if the user provided it?

```suggestion
            experiment = experiment_name or f"{get_one_of('FLYTE_INTERNAL_EXECUTION_WORKFLOW', '_F_WF')}"
            run_name = f"{params.execution_id.name}.{params.task_id.name.split('.')[-1]}"
```

So basically use the execution name (not sure what `_F_WF` is exactly) if the user didn't explicitly specify an `experiment_name` in the decorator.

Reply:
yes, updated it.  `_F_WF` is the [default workflow name](https://github.com/flyteorg/flytekit/blob/0dcda1dda1290077cbaf5b6143e0f226c4e9142a/flytekit/bin/entrypoint.py#L199) if flytekit can't find the env `FLYTE_INTERNAL_EXECUTION_WORKFLOW`